backend-api/src/functions/asyncActiveSession/tokenService/publicKeyGetter.ts

```diff
@@ -1,40 +1,53 @@
-import {
-  decodeProtectedHeader,
-  importJWK,
-  KeyLike,
-  ProtectedHeaderParameters,
-} from "jose";
+import { importJWK, KeyLike } from "jose";
 import { errorResult, Result, successResult } from "../../utils/result";
+import {
+  ISendHttpRequest,
+  sendHttpRequest,
+} from "../../services/http/sendHttpRequest";
+
+export interface IPublicKeyGetter {
+  getPublicKey: (
+    jwksEndpoint: string,
+    kid: string,
+  ) => Promise<Result<KeyLike | Uint8Array>>;
+}
+
+export interface IJwks {
+  keys: IJwk[];
+}
+
+export interface IJwk extends JsonWebKey {
+  alg: "ES256";
+  kid: string;
+  kty: "EC";
+  use: "sig";
+}
+
+export type PublicKeyGetterDependencies = {
+  sendHttpRequest: ISendHttpRequest;
+};
+
+const publicKeyGetterDependencies: PublicKeyGetterDependencies = {
+  sendHttpRequest: sendHttpRequest,
+};
 
 export class PublicKeyGetter implements IPublicKeyGetter {
+  private readonly sendHttpRequest: ISendHttpRequest;
+  constructor(dependencies = publicKeyGetterDependencies) {
+    this.sendHttpRequest = dependencies.sendHttpRequest;
+  }
+
   async getPublicKey(
-    jwks: IJwks,
-    jwt: string,
+    jwksEndpoint: string,
+    kid: string,
   ): Promise<Result<Uint8Array | KeyLike>> {
-    let decodedProtectedHeader: ProtectedHeaderParameters;
+    let jwk;
     try {
-      decodedProtectedHeader = decodeProtectedHeader(jwt);
```

Comment:
I've moved this header logic to the TokenService. It made more sense there and it facilitated returning a client error when the header can't be decoded or there is no kid claim.